File: apps/backgammon/backend/app/tiers.py
"""League tier mapping for ranked play.

Maps a player's ELO rating to one of five tier buckets. Used on leaderboard,
dashboard, and player profile responses.

Thresholds are intentionally coarse and exclusive upper-bounds so the function
is a pure mapping (no hysteresis / promotion logic yet).
"""

from typing import Final


# Ordered from lowest to highest.
TIER_ORDER: Final[tuple[str, ...]] = (
    "Bronze",
    "Silver",
    "Gold",
    "Platinum",
    "Diamond",
)


def tier_for_rating(rating: int) -> str:
    """Return the tier name for a given ELO rating.

    - Bronze:   <1400
    - Silver:   1400-1599
    - Gold:     1600-1799
    - Platinum: 1800-1999
    - Diamond:  >=2000
    """
    if rating < 1400:
        return "Bronze"
    if rating < 1600:
        return "Silver"
    if rating < 1800:
        return "Gold"
    if rating < 2000:
        return "Platinum"
    return "Diamond"

File: apps/backgammon/frontend/src/constants/tiers.ts
/**
 * League tier mapping for ranked play.
 *
 * Mirrors `backend/app/tiers.py`. Tier is derived from the player's ELO
 * rating using exclusive upper-bound thresholds.
 */

export type Tier = "Bronze" | "Silver" | "Gold" | "Platinum" | "Diamond";

export const TIER_ORDER: readonly Tier[] = [
  "Bronze",
  "Silver",
  "Gold",
  "Platinum",
  "Diamond",
];

/** Return the tier for a given ELO rating. */
export function tierForRating(rating: number): Tier {
  if (rating < 1400) return "Bronze";
  if (rating < 1600) return "Silver";
  if (rating < 1800) return "Gold";
  if (rating < 2000) return "Platinum";
  return "Diamond";
}

/** Colour used for the tier badge; uses CSS variables with a few tier-specific hues. */
export const TIER_COLORS: Record<Tier, string> = {
  Bronze: "#a97449",
  Silver: "#b9b9c4",
  Gold: "#d4a843",
  Platinum: "#8ce3d5",
  Diamond: "#7ec4ff",
};
